src/app/application/services/data-source/DataSourcePool.ts

```diff
@@ -0,0 +1,80 @@
+import CallableEventSource, { isClosed } from './CallableEventSource'
+import Router from './Router'
+import SharedPool from './SharedPool'
+
+export type DataSourceResponse<T> = {data: T | undefined, error: Error | undefined, refresh: () => void}
+export type Source = (params: Record<string, unknown>, source: {close: () => void}) => Promise<unknown>
+export type Sources = Record<string, Source>
+
+const create = (src: string, router: Router<Source>): CallableEventSource => {
+  const [path, query] = src.split('?')
+  const queryParams = new URLSearchParams(query)
+  const route = router.match(path)
+  const _source = new CallableEventSource(async function * (this: CallableEventSource) {
+    while (true) {
+      this.readyState = 1
+      // latency
+      // await new Promise(resolve => setTimeout(resolve, 5000))
+      yield route.route({
+        ...{
+          offset: parseInt(queryParams.get('offset') || '0'),
```

Comment:
Why is offset receiving special treatment here?

Reply:
If you mean the `parseInt`, it's just to turn it into a number for the argument for the function call/HTTP API request in the `sources` file. I was thinking by the time you get to the function call offset should be a number.

Slightly more longer term I think I would rather specify a `page` and then calculate the correct offset behind the scenes (this is how I do it in the original code for this). But I didn't want to bring to much in here rn.

Actually you've just reminded me, I think we said we would rename `route.route` to something else so its clearer that we are calling the `'/uri/:arg' => Promise` here.

--

(Let me know if I misunderstood the question here)